tests/ha/single_pod_restart/test_server_pod_restart_api.py

```diff
@@ -768,3 +793,198 @@ def test_copy_obj_after_server_pod_restart(self):
+                    "the same.", self.object_name)
+        LOGGER.info("Step 4: Shutdown server pod with replica method and verify cluster & "
+                    "remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            pod_prefix=[const.SERVER_POD_NAME_PREFIX], delete_pod=[self.delete_pod],
+            num_replica=self.num_replica-1)
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete server pod")
+        pod_name = list(resp[1].keys())[0]
+        if self.set_type == const.STATEFULSET:
+            self.set_name = resp[1][pod_name]['deployment_name']
+        elif self.set_type == const.REPLICASET:
+            self.deployment_name = resp[1][pod_name]['deployment_name']
+        self.restore_pod = True
+        self.restore_method = resp[1][pod_name]['method']
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 4: Successfully shutdown server pod %s. Verified cluster and services "
+                    "states are as expected & remaining pods status is online.", pod_name)
+        LOGGER.info("Step 5: Get object versions of %s with VersionID & verify etags.",
+                    self.object_name)
+        resp = self.ha_obj.parallel_get_object(event=event, s3_ver_obj=self.s3_ver,
+                                               bkt_name=self.bucket_name, obj_name=self.object_name,
+                                               ver_etag=self.version_etag[self.bucket_name])
+        assert_utils.assert_true(resp[0], f"Get Object with versionID failed {resp[1]}")
+        LOGGER.info("Step 5: Got object versions of %s with VersionID & verified etags.",
+                    self.object_name)
+        count = HA_CFG["common_params"]["put_get_version"]
+        LOGGER.info("Step 6: Starting background threads for Get and Put Version for count %s "
+                    "while server pod restarting.", count)
+        LOGGER.info("Upload same object %s for %s times for background Get.", self.object_name,
+                    count)
+        args = {'file_path': self.multipart_obj_path, 'count': count}
+        resp = self.ha_obj.parallel_put_object(event, self.s3_test_obj, self.bucket_name,
+                                               self.object_name, **args)
+        assert_utils.assert_true(resp[0], f"Upload Object failed {resp[1]}")
+        self.version_etag[self.bucket_name].extend(resp[1])
+        args = {'file_path': self.multipart_obj_path, 'count': count, 'background': True}
+        put_thread = threading.Thread(
+            target=self.ha_obj.parallel_put_object,
+            args=(event, self.s3_test_obj, self.bucket_name, self.object_name, put_output),
+            kwargs=args)
+        args = {'background': True}
+        get_thread = threading.Thread(
```

Comment:
so if I understand this right, we will just getting object which was done before restart and not for the one we are putting in parallel?

Reply:
> so if I understand this right, we will just getting object which was done before restart and not for the one we are putting in parallel?

Yes. Similar like what we do in READs/WRITEs parallel, as we don't want to fail GET in case if PUT itself fails.
2 different threads working parallel to check both PUT and GET functionality of versioning.